code/modules/mob/living/simple_animal/hostile/mining/goldgrub.dm

```diff
@@ -42,7 +42,7 @@
 	if(target != null)
 		if(istype(target, /obj/item/stack/ore) && loot.len < 10)
 			visible_message("<span class='notice'>The [name] looks at [target.name] with hungry eyes.</span>")
-		else if(isliving(target))
+		else if(isliving(target) || istype(target, /obj/mecha))
```

Comment:
else if(isliving(target) || ismecha(target))

Reply:
That's much better thanks